helix-term/src/ui/completion.rs

```diff
@@ -82,18 +87,101 @@ impl menu::Item for CompletionItem {
                     ""
                 }
                 None => "",
-            }),
+            },
+            CompletionItem::Path(PathCompletionItem { kind, .. }) => kind.as_str(),
+        };
+
+        menu::Row::new(vec![
+            menu::Cell::from(Span::styled(
+                label,
+                if deprecated {
+                    Style::default().add_modifier(Modifier::CROSSED_OUT)
+                } else {
+                    Style::default()
+                },
+            )),
+            menu::Cell::from(kind),
         ])
     }
 }
 
-#[derive(Debug, PartialEq, Default, Clone)]
-pub struct CompletionItem {
+#[derive(Debug, PartialEq, Clone)]
+pub enum PathKind {
+    Folder,
+    File,
+    Link,
+    Block,
+    Socket,
+    CharacterDevice,
+    Fifo,
+}
+
+impl PathKind {
+    fn as_str(&self) -> &'static str {
+        match self {
+            PathKind::Folder => "folder",
+            PathKind::File => "file",
+            PathKind::Link => "link",
+            PathKind::Block => "block",
+            PathKind::Socket => "socket",
+            PathKind::CharacterDevice => "char_device",
+            PathKind::Fifo => "fifo",
+        }
+    }
+}
+
+impl std::fmt::Display for PathKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.as_str())
+    }
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct LspCompletionItem {
     pub item: lsp::CompletionItem,
     pub provider: LanguageServerId,
     pub resolved: bool,
 }
 
+#[derive(Debug, PartialEq, Clone)]
+pub struct PathCompletionItem {
+    pub kind: PathKind,
+    pub item: helix_core::CompletionItem,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub enum CompletionItem {
+    Lsp(LspCompletionItem),
+    Path(PathCompletionItem),
+}
+
+impl PartialEq<CompletionItem> for LspCompletionItem {
```

Comment:
I don't think we necessarily need these

Reply:
See https://github.com/helix-editor/helix/pull/2608#discussion_r1740929468